### DISCUSSION TITLE
Developing A Cask For pre-release software

### DISCUSSION THREAD

**Discussion by tmrdev** [Homebrew/discussions#4656] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.28-74-g8ffac3e
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8ffac3e2ac680bd8167e85390e843950bd3b63e3
Last commit: 2 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: aa410805836c56bfebd2d856eab60a1ef9f0c886
Core tap last commit: 5 months ago
Core tap branch: master
Core tap JSON: 17 Jul 06:11 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: octa-core 64-bit kabylake
Clang: 12.0.5 build 1205
Git: 2.30.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.6-x86_64
CLT: 13.2.0.0.1.1638488800
Xcode: 12.5
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  hping

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  openssl@3

Warning: You have uncommitted modifications to Homebrew/homebrew-cask.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask && git stash -u && git clean -d -f

Uncommitted files:
  ?? Casks/.greenery.rb.swp

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/cask) checkout master

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> /Users/sonicbur/.bash_profile

Warning: Your Xcode (12.5) is outdated.
Please update to Xcode 13.2.1 (or delete it).
Xcode can be updated from the App Store.
```

### Description of issue

I am trying to submit a GUI app called Greenery, there will not be a CLI component for Greenery so all I need to do is submit my Cask file, correct?  If the app is GUI only then a Formulae is not needed?

Our app is in a pre-release state at 0.9.0 and we may have a release candidate in about two to three months.  Is there a proper Cask configuration for pre-release software?

I have forked the official homebrew-cask repo to my personal account here: https://github.com/tmrdev/homebrew-cask/tree/greenery

I created a branch called greenery which is also the name of my cask.  It looks like my PR was submitted but I need to double check that.

I also was testing a 1.0 version of the Greenery app to see if I could get them to work and I get the following with the cask audit command: 

brew audit --new-cask greenery

audit for greenery: failed
 - Version '0.9.0' differs from '1.0' retrieved by livecheck.
 - Version '0.9.0' differs from '1.0' retrieved by livecheck.
 - v0.9.0 is a GitHub pre-release.
 - GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)

Thanks in advance for any help with these issues.

**Comment by SMillerDev** [comment]
> Our app is in a pre-release state at 0.9.0 and we may have a release candidate in about two to three months. Is there a proper Cask configuration for pre-release software?

Homebrew as a whole prefers not to ship pre-releases. Your software can wait for inclusion until it's stable.

> Version '0.9.0' differs from '1.0' retrieved by livecheck.

Your repo has a 1.0 tag, that is semantically higher than the version in your cask. Why include software that is not the latest version?

> GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)

https://docs.brew.sh/Acceptable-Casks#rejected-casks

**Reply by SMillerDev** [reply]
Additionally, please don't email individual maintainers. Someone will respond when they want to, it's really rude to demand volunteers spend their time on you like that.

**Reply by tmrdev** [reply]
I referenced this thread in my email to you. I was not demanding that you respond to it.  I'm sorry that you took offense to the email.

In regards to the apps pre-release state, the app is functional and stable.  The UI needs more work and we are close to a 1.0.0 release.

As for the 1.0 tag, I was testing a 1.0 version of the app on the relevant github repo.  I removed the 1.0 version from the github repo but the brew audit commands still show it as being active.  I need to do more testing and I believe the 1.0 version is cached (I see a local cache on macOS).

brew doctor shows this:

* Warning: You have uncommitted modifications to Homebrew/homebrew-cask.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.

  cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask && git stash -u && git clean -d -f

I'm going to look into stashing my changes and starting over.

**Comment by tlk** [comment, marked as answer]
Check out https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap#creating-a-tap - if you create a "custom tap", your users can install your software like this `brew install xyz/repo/yoursoftware` where xyz is your github username. You do not need approval or permission from homebrew maintainers if you go in this direction.
